<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>High Performance GWAS Plotting And Annotation • gwasplot</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="High Performance GWAS Plotting And Annotation">
<meta name="description" content="More about what it does (maybe more than one line). Continuation lines should be indented.">
<meta property="og:description" content="More about what it does (maybe more than one line). Continuation lines should be indented.">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">gwasplot</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/weinstocklab/gwasplot/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h1></div>
<p><strong>gwasplot</strong> provides fast and efficient tools for visualizing and annotating GWAS summary statistics. It offers functions to reformat data, create plots, and annotate top hits with genomic features.</p>
<p>It is designed to work with output from GWAS performed on WGS data with many rare variants, using <a href="https://duckdb.org/" class="external-link">duckdb</a> to manipulate data.</p>
<p>For more information, visit the <a href="https://github.com/weinstockj/gwasplot" class="external-link">GitHub repository</a> or the <a href="https://weinstockj.github.io/gwasplot/" class="external-link">project website</a>.</p>
<div class="section level2">
<h2 id="roadmap">Roadmap<a class="anchor" aria-label="anchor" href="#roadmap"></a>
</h2>
<ul>
<li>add LD into locuszoom plots, perhaps referencing this <a href="https://github.com/weinstockj/LD_REFERENCE_PANEL" class="external-link">setup</a>.</li>
<li>add in helper functions to faciliate fine-mapping with other packages (e.g., SuSIE)</li>
</ul>
</div>
</div>
<div class="section level1">
<h1 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h1>
<p>Install the development version of <strong>gwasplot</strong> from GitHub:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Install remotes if not already installed</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"remotes"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"remotes"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Install gwasplot from GitHub</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"weinstockj/gwasplot"</span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="input-formats">Input formats<a class="anchor" aria-label="anchor" href="#input-formats"></a>
</h2>
<p><code>gwasplot</code> expects output from either regenie or saige in csv or parquet format (parquet is generally recommended).</p>
<p>Regenie output looks like this:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>dplyr<span class="sc">::</span><span class="fu">glimpse</span>(df)</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>Rows<span class="sc">:</span> <span class="dv">5</span></span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a>Columns<span class="sc">:</span> <span class="dv">12</span></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="sc">$</span> CHROM     <span class="sc">&lt;</span>chr<span class="sc">&gt;</span> <span class="st">"chr3"</span>, <span class="st">"chr3"</span>, <span class="st">"chr3"</span>, <span class="st">"chr3"</span>, <span class="st">"chr3"</span></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="sc">$</span> POS       <span class="sc">&lt;</span>int<span class="sc">&gt;</span> <span class="dv">90000011</span>, <span class="dv">90000261</span>, <span class="dv">90000274</span>, <span class="dv">90000324</span>, <span class="dv">90000366</span></span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a><span class="sc">$</span> ID        <span class="sc">&lt;</span>chr<span class="sc">&gt;</span> <span class="st">"chr3-90000011-T-C"</span>, <span class="st">"chr3-90000261-GATT-G"</span>, <span class="st">"chr3-900002…"</span></span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a><span class="sc">$</span> ALLELE0   <span class="sc">&lt;</span>chr<span class="sc">&gt;</span> <span class="st">"T"</span>, <span class="st">"GATT"</span>, <span class="st">"T"</span>, <span class="st">"C"</span>, <span class="st">"T"</span></span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a><span class="sc">$</span> ALLELE1   <span class="sc">&lt;</span>chr<span class="sc">&gt;</span> <span class="st">"C"</span>, <span class="st">"G"</span>, <span class="st">"A"</span>, <span class="st">"A"</span>, <span class="st">"G"</span></span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a><span class="sc">$</span> A1FREQ    <span class="sc">&lt;</span>dbl<span class="sc">&gt;</span> <span class="fl">1.37775e-04</span>, <span class="fl">2.91090e-04</span>, <span class="fl">2.38259e-04</span>, <span class="fl">1.62637e-04</span>, <span class="fl">4.66158</span>e…</span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a><span class="sc">$</span> N         <span class="sc">&lt;</span>int<span class="sc">&gt;</span> <span class="dv">482670</span>, <span class="dv">482669</span>, <span class="dv">482669</span>, <span class="dv">482669</span>, <span class="dv">482669</span></span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a><span class="sc">$</span> BETA      <span class="sc">&lt;</span>dbl<span class="sc">&gt;</span> <span class="sc">-</span><span class="fl">0.00157451</span>, <span class="sc">-</span><span class="fl">0.06463220</span>, <span class="sc">-</span><span class="fl">0.06452010</span>, <span class="fl">0.02796820</span>, <span class="sc">-</span><span class="fl">0.021063</span>…</span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a><span class="sc">$</span> SE        <span class="sc">&lt;</span>dbl<span class="sc">&gt;</span> <span class="fl">0.0730887</span>, <span class="fl">0.0479369</span>, <span class="fl">0.0540183</span>, <span class="fl">0.0646778</span>, <span class="fl">0.1213880</span></span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a><span class="sc">$</span> CHISQ     <span class="sc">&lt;</span>dbl<span class="sc">&gt;</span> <span class="fl">0.000464077</span>, <span class="fl">1.817850000</span>, <span class="fl">1.426620000</span>, <span class="fl">0.186990000</span>, <span class="fl">0.030108</span>…</span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a><span class="sc">$</span> LOG10P    <span class="sc">&lt;</span>dbl<span class="sc">&gt;</span> <span class="fl">0.00752913</span>, <span class="fl">0.75063100</span>, <span class="fl">0.63391900</span>, <span class="fl">0.17689500</span>, <span class="fl">0.06437000</span></span></code></pre></div>
<p>gwastools will read in data from either regenie or saige and standarize the column names, and then create a table called <code>summary_stats</code> in a <code>duckdb</code> database.</p>
</div>
</div>
<div class="section level1">
<h1 id="key-features">Key Features<a class="anchor" aria-label="anchor" href="#key-features"></a>
</h1>
<div class="section level2">
<h2 id="data-quality-control-and-filtering">Data Quality Control and Filtering<a class="anchor" aria-label="anchor" href="#data-quality-control-and-filtering"></a>
</h2>
<p><strong>gwasplot</strong> includes robust filtering capabilities to remove variants from problematic genomic regions that can lead to spurious associations:</p>
<div class="section level3">
<h3 id="exclude-difficult-regions">Exclude Difficult Regions<a class="anchor" aria-label="anchor" href="#exclude-difficult-regions"></a>
</h3>
<p>The <code><a href="reference/exclude_difficult_regions.html">exclude_difficult_regions()</a></code> function removes variants from regions with known mapping difficulties:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Remove variants from multiple types of difficult regions</span></span>
<span><span class="va">gwas_filtered</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/exclude_difficult_regions.html">exclude_difficult_regions</a></span><span class="op">(</span><span class="va">gwas_stats</span>, </span>
<span>                                         beds_exclude <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"hg19diff"</span>, <span class="st">"UCSC_unusual"</span>, <span class="st">"GRC_exclusions"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Available region types include: - <strong>hg19diff</strong>: Regions with assembly differences between genome builds - <strong>UCSC_unusual</strong>: Unusual regions identified by UCSC (gaps, heterochromatin, etc.) - <strong>GRC_exclusions</strong>: Genome Reference Consortium exclusion regions - <strong>GIAB_difficult_regions</strong>: Genome in a Bottle difficult-to-map regions</p>
</div>
<div class="section level3">
<h3 id="custom-variant-filtering">Custom Variant Filtering<a class="anchor" aria-label="anchor" href="#custom-variant-filtering"></a>
</h3>
<p>Filter variants using custom whitelists or blacklists:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Keep only variants in a specific region or set</span></span>
<span><span class="va">filtered_gwas</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/filter_variants.html">filter_variants</a></span><span class="op">(</span><span class="va">gwas_stats</span>, subset <span class="op">=</span> <span class="st">"my_variants.parquet"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Exclude specific variants</span></span>
<span><span class="va">filtered_gwas</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/filter_variants.html">filter_variants</a></span><span class="op">(</span><span class="va">gwas_stats</span>, exclude <span class="op">=</span> <span class="st">"bad_variants.parquet"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="gene-annotation">Gene Annotation<a class="anchor" aria-label="anchor" href="#gene-annotation"></a>
</h2>
<p>Annotate variants with nearest protein-coding genes:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Add gene annotations</span></span>
<span><span class="va">annotated_gwas</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/find_nearest_gene.html">find_nearest_gene</a></span><span class="op">(</span><span class="va">gwas_stats</span>, threshold <span class="op">=</span> <span class="fl">1e5</span><span class="op">)</span>  <span class="co"># 100kb window</span></span></code></pre></div>
<p>Additional specialized annotations: - <strong>CHIP genes</strong>: <code><a href="reference/annotate_with_chip_genes.html">annotate_with_chip_genes()</a></code> - flags variants in clonal hematopoiesis genes - <strong>Centromeres</strong>: <code><a href="reference/annotate_with_centromere.html">annotate_with_centromere()</a></code> - identifies variants in centromeric regions<br>
- <strong>Immunoglobulin loci</strong>: <code><a href="reference/annotate_with_immunoglobulin.html">annotate_with_immunoglobulin()</a></code> - flags variants in Ig heavy/light chain regions</p>
</div>
<div class="section level2">
<h2 id="visualization">Visualization<a class="anchor" aria-label="anchor" href="#visualization"></a>
</h2>
<div class="section level3">
<h3 id="manhattan-plots">Manhattan Plots<a class="anchor" aria-label="anchor" href="#manhattan-plots"></a>
</h3>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/manhattan.html">manhattan</a></span><span class="op">(</span>gwas <span class="op">=</span> <span class="va">gwas_stats</span>, output_prefix <span class="op">=</span> <span class="st">"my_manhattan"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="qq-plots">QQ Plots<a class="anchor" aria-label="anchor" href="#qq-plots"></a>
</h3>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/qqplot_save.html">qqplot_save</a></span><span class="op">(</span>gwas <span class="op">=</span> <span class="va">gwas_stats</span>, output_prefix <span class="op">=</span> <span class="st">"my_qqplot"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="locuszoom-plots">LocusZoom Plots<a class="anchor" aria-label="anchor" href="#locuszoom-plots"></a>
</h3>
<p>Create detailed regional association plots with gene tracks:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot a specific locus with gene structure</span></span>
<span><span class="fu"><a href="reference/locuszoom.html">locuszoom</a></span><span class="op">(</span><span class="va">gwas_stats</span>, locus_chr <span class="op">=</span> <span class="st">"chr2"</span>, locus_start <span class="op">=</span> <span class="fl">25000000</span>, locus_end <span class="op">=</span> <span class="fl">25500000</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="volcano-plots">Volcano Plots<a class="anchor" aria-label="anchor" href="#volcano-plots"></a>
</h3>
<p>For post-GWAS analysis (e.g., after empirical Bayes shrinkage):</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Automatically uses lfsr if available, otherwise p-values</span></span>
<span><span class="fu"><a href="reference/volcano.html">volcano</a></span><span class="op">(</span><span class="va">gwas_data</span>, phenotype_label <span class="op">=</span> <span class="st">"My Phenotype"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
</div>
<div class="section level1">
<h1 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h1>
<p>Below are some sample usage examples:</p>
<div class="section level2">
<h2 id="basic-workflow">Basic Workflow<a class="anchor" aria-label="anchor" href="#basic-workflow"></a>
</h2>
<p>Reformat GWAS Summary Statistics Use the reformat_summary_statistics function to read and reformat GWAS summary statistics from a parquet or CSV file:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="fu">library</span>(gwasplot)</span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a></span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a><span class="co"># Reformat summary statistics from a file</span></span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a>gwas_stats <span class="ot">&lt;-</span> <span class="fu">reformat_summary_statistics</span>(<span class="st">"path/to/your/file.parquet"</span>)</span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a><span class="fu">print</span>(gwas_stats)</span>
<span id="cb10-6"><a href="#cb10-6" tabindex="-1"></a></span>
<span id="cb10-7"><a href="#cb10-7" tabindex="-1"></a>GWAS object</span>
<span id="cb10-8"><a href="#cb10-8" tabindex="-1"></a>File path<span class="sc">:</span> ..<span class="sc">/</span>concatenated_results.parquet</span>
<span id="cb10-9"><a href="#cb10-9" tabindex="-1"></a>Detected format<span class="sc">:</span> regenie</span>
<span id="cb10-10"><a href="#cb10-10" tabindex="-1"></a>Data names<span class="sc">:</span> CHROM, POS, ID, ALLELE0, ALLELE1, A1FREQ, N, BETA, SE, CHISQ, LOG10P, phenotype</span>
<span id="cb10-11"><a href="#cb10-11" tabindex="-1"></a>Data preview<span class="sc">:</span></span>
<span id="cb10-12"><a href="#cb10-12" tabindex="-1"></a><span class="co"># A tibble: 5 × 9</span></span>
<span id="cb10-13"><a href="#cb10-13" tabindex="-1"></a>  CHROM      POS REF   ALT      AF_ALT     BETA  LOG10P PVALUE ID</span>
<span id="cb10-14"><a href="#cb10-14" tabindex="-1"></a>  <span class="sc">&lt;</span>chr<span class="sc">&gt;</span>    <span class="er">&lt;</span>dbl<span class="sc">&gt;</span> <span class="er">&lt;</span>chr<span class="sc">&gt;</span> <span class="er">&lt;</span>chr<span class="sc">&gt;</span>     <span class="er">&lt;</span>dbl<span class="sc">&gt;</span>    <span class="er">&lt;</span>dbl<span class="sc">&gt;</span>   <span class="er">&lt;</span>dbl<span class="sc">&gt;</span>  <span class="er">&lt;</span>dbl<span class="sc">&gt;</span> <span class="er">&lt;</span>chr<span class="sc">&gt;</span></span>
<span id="cb10-15"><a href="#cb10-15" tabindex="-1"></a><span class="dv">1</span> chr3  <span class="dv">90000011</span> T     C     <span class="fl">0.000138</span>  <span class="sc">-</span><span class="fl">0.00157</span> <span class="fl">0.00753</span>  <span class="fl">0.983</span> chr3_90000011_T_C</span>
<span id="cb10-16"><a href="#cb10-16" tabindex="-1"></a><span class="dv">2</span> chr3  <span class="dv">90000261</span> GATT  G     <span class="fl">0.000291</span>  <span class="sc">-</span><span class="fl">0.0646</span>  <span class="fl">0.751</span>    <span class="fl">0.178</span> chr3_90000261_GA…</span>
<span id="cb10-17"><a href="#cb10-17" tabindex="-1"></a><span class="dv">3</span> chr3  <span class="dv">90000274</span> T     A     <span class="fl">0.000238</span>  <span class="sc">-</span><span class="fl">0.0645</span>  <span class="fl">0.634</span>    <span class="fl">0.232</span> chr3_90000274_T_A</span>
<span id="cb10-18"><a href="#cb10-18" tabindex="-1"></a><span class="dv">4</span> chr3  <span class="dv">90000324</span> C     A     <span class="fl">0.000163</span>   <span class="fl">0.0280</span>  <span class="fl">0.177</span>    <span class="fl">0.665</span> chr3_90000324_C_A</span>
<span id="cb10-19"><a href="#cb10-19" tabindex="-1"></a><span class="dv">5</span> chr3  <span class="dv">90000366</span> T     G     <span class="fl">0.0000466</span> <span class="sc">-</span><span class="fl">0.0211</span>  <span class="fl">0.0644</span>   <span class="fl">0.862</span> chr3_90000366_T_G</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="complete-analysis-pipeline">Complete Analysis Pipeline<a class="anchor" aria-label="anchor" href="#complete-analysis-pipeline"></a>
</h2>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load and reformat data</span></span>
<span><span class="va">gwas_stats</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/reformat_summary_statistics.html">reformat_summary_statistics</a></span><span class="op">(</span><span class="st">"path/to/gwas_results.parquet"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Quality control: remove problematic regions for WGS GWAS</span></span>
<span><span class="va">gwas_clean</span> <span class="op">&lt;-</span> <span class="va">gwas_stats</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="reference/exclude_difficult_regions.html">exclude_difficult_regions</a></span><span class="op">(</span></span>
<span>    beds_exclude <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>      <span class="st">"hg19diff"</span>, </span>
<span>      <span class="st">"UCSC_unusual"</span>, </span>
<span>      <span class="st">"GRC_exclusions"</span></span>
<span>      <span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="fu"><a href="reference/manhattan.html">manhattan</a></span><span class="op">(</span><span class="va">gwas_clean</span>, output <span class="op">=</span> <span class="st">"my_analysis.manhattan.pdf"</span><span class="op">)</span></span>
<span><span class="fu"><a href="reference/qqplot.html">qqplot</a></span><span class="op">(</span><span class="va">gwas_clean</span>, output <span class="op">=</span> <span class="st">"my_analysis.qq.pdf"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get top hits and annotate with genes</span></span>
<span><span class="va">top_hits</span> <span class="op">&lt;-</span> <span class="va">gwas_clean</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="reference/select_top_hits.html">select_top_hits</a></span><span class="op">(</span>threshold <span class="op">=</span> <span class="fl">5e-8</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="reference/find_nearest_gene.html">find_nearest_gene</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot specific loci</span></span>
<span><span class="fu"><a href="reference/locuszoom.html">locuszoom</a></span><span class="op">(</span><span class="va">gwas_clean</span>, locus_chr <span class="op">=</span> <span class="st">"chr2"</span>, locus_start <span class="op">=</span> <span class="fl">25000000</span>, locus_end <span class="op">=</span> <span class="fl">25500000</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="contact">Contact<a class="anchor" aria-label="anchor" href="#contact"></a>
</h2>
<p>Email Josh Weinstock. See our other work <a href="https://weinstocklab.org" class="external-link">here</a>.</p>
</div>
</div>

  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/weinstocklab/gwasplot/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/weinstocklab/gwasplot/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small>GPL (&gt;= 3)</small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing gwasplot</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Josh Weinstock <br><small class="roles"> Author, maintainer </small>   </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Josh Weinstock.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
