<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Package index â€¢ gwasplot</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Package index"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">gwasplot</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/weinstockj/gwasplot/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Package index</h1>
    </div>

    <div class="section level2">
      <h2 id="all-functions">All functions<a class="anchor" aria-label="anchor" href="#all-functions"></a></h2>




    </div><div class="section level2">




      <dl><dt>

          <code><a href="annotate_variants_ensembl.html">annotate_variants_ensembl()</a></code>

        </dt>
        <dd>Annotate variants with functional consequences using Ensembl VEP API</dd>
      </dl><dl><dt>

          <code><a href="annotate_with_centromere.html">annotate_with_centromere()</a></code>

        </dt>
        <dd>Annotate data with centromere information</dd>
      </dl><dl><dt>

          <code><a href="annotate_with_chip_genes.GWASFormatter.html">annotate_with_chip_genes(<i>&lt;GWASFormatter&gt;</i>)</a></code>

        </dt>
        <dd>Annotate top hits with CHIP gene information</dd>
      </dl><dl><dt>

          <code><a href="annotate_with_chip_genes.html">annotate_with_chip_genes()</a></code>

        </dt>
        <dd>Annotate data with CHIP gene information</dd>
      </dl><dl><dt>

          <code><a href="annotate_with_immunoglobulin.html">annotate_with_immunoglobulin()</a></code>

        </dt>
        <dd>Annotate data with immunoglobulin gene information</dd>
      </dl><dl><dt>

          <code><a href="append_ashr_results.html">append_ashr_results()</a></code>

        </dt>
        <dd>Append lfsr results from ashr to a data frame</dd>
      </dl><dl><dt>

          <code><a href="assign_gene_tracks.html">assign_gene_tracks()</a></code>

        </dt>
        <dd>Assign genes to tracks to avoid overlaps</dd>
      </dl><dl><dt>

          <code><a href="compute_sample.html">compute_sample()</a></code>

        </dt>
        <dd>Take a random sample of rows from the summary stats table</dd>
      </dl><dl><dt>

          <code><a href="db_connect.html">db_connect()</a></code>

        </dt>
        <dd>Connect to DuckDB</dd>
      </dl><dl><dt>

          <code><a href="exclude_difficult_regions.html">exclude_difficult_regions()</a></code>

        </dt>
        <dd>Exclude difficult regions</dd>
      </dl><dl><dt>

          <code><a href="filter_variants.html">filter_variants()</a></code>

        </dt>
        <dd>Filter variants</dd>
      </dl><dl><dt>

          <code><a href="find_nearest_gene.html">find_nearest_gene()</a></code>

        </dt>
        <dd>Find the nearest gene for variants</dd>
      </dl><dl><dt>

          <code><a href="get_biosample_for_accession.html">get_biosample_for_accession()</a></code>

        </dt>
        <dd>Get biosample to accession mapping for a specific accession</dd>
      </dl><dl><dt>

          <code><a href="get_encode_biosamples.html">get_encode_biosamples()</a></code>

        </dt>
        <dd>Get available ENCODE SCREEN biosamples for a given assembly</dd>
      </dl><dl><dt>

          <code><a href="locuszoom.html">locuszoom()</a></code>

        </dt>
        <dd>LocusZoom-style plot for GWAS results</dd>
      </dl><dl><dt>

          <code><a href="manhattan.html">manhattan()</a></code>

        </dt>
        <dd>Plot a Manhattan plot from a gwas object</dd>
      </dl><dl><dt>

          <code><a href="qqplot.html">qqplot()</a></code>

        </dt>
        <dd>QQ plot for p-values</dd>
      </dl><dl><dt>

          <code><a href="qqplot_save.html">qqplot_save()</a></code>

        </dt>
        <dd>Save a QQ plot of GWAS p-values</dd>
      </dl><dl><dt>

          <code><a href="query_encode_ccres.html">query_encode_ccres()</a></code>

        </dt>
        <dd>Query ENCODE SCREEN cCREs for a genomic region</dd>
      </dl><dl><dt>

          <code><a href="query_ot_api_v2g.html">query_ot_api_v2g()</a></code>

        </dt>
        <dd>Query Open Targets Platform API for Locus-to-Gene (L2G) predictions</dd>
      </dl><dl><dt>

          <code><a href="query_ot_api_variants.html">query_ot_api_variants()</a></code>

        </dt>
        <dd>Query Open Targets Platform API for variant information</dd>
      </dl><dl><dt>

          <code><a href="reformat_names.html">reformat_names()</a></code>

        </dt>
        <dd>Rename columns in the dataset to standard names</dd>
      </dl><dl><dt>

          <code><a href="reformat_summary_statistics.html">reformat_summary_statistics()</a></code>

        </dt>
        <dd>Reformat GWAS summary statistics from regenie or SAIGE</dd>
      </dl><dl><dt>

          <code><a href="search_cell_types.html">search_cell_types()</a></code>

        </dt>
        <dd>Search for available cell types and their biosample IDs</dd>
      </dl><dl><dt>

          <code><a href="select_top_hits.html">select_top_hits()</a></code>

        </dt>
        <dd>Pull top hits from a GWASFormatter or a data.frame/tibble</dd>
      </dl><dl><dt>

          <code><a href="volcano.html">volcano()</a></code>

        </dt>
        <dd>Volcano plot for GWAS results</dd>
      </dl></div>
  </main></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Josh Weinstock.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

